src/linear_samplers/block_col_SRHT.jl

```diff
@@ -0,0 +1,85 @@
+
+"""
+    LinSysBlockColSRHT <: LinSysBlkColSampler
+
+A mutable structure with fields to handle SRHT column sketching. For this procedure,
+the hadamard transform and random sign swaps are applied once, then that matrix is repeatably
+sampled.
+
+# Fields
+- `block_size::Int64`, the size of blocks being chosen
+- `padded_size::Int64`, the size of the matrix when padded
+- `block::Union{Vector{Int64}, Nothing}`, storage for block indices
+- `hadamard::Union{AbstractMatrix, Nothing}`, storage for the hadamard matrix.
+- `Ap::Union{AbstractMatrix, Nothing}`, storage for padded matrix
+- `signs::Union{Vector{Bool}, Nothing}`, storage for random sign flips.
+- `scaling::Float64`, storage for the scaling of the sketches.
+
+Calling `LinSysBlockColSRHT()` defaults to setting `block_size` to 2.
+
+Nir Ailon and Bernard Chazelle. 2006. Approximate nearest neighbors and the fast Johnson-Lindenstrauss transform. In Proceedings of the thirty-eighth annual ACM symposium on Theory of Computing (STOC '06). Association for Computing Machinery, New York, NY, USA, 557–563. https://doi.org/10.1145/1132516.1132597
```

Comment:
Wrong reference?

Reply:
This is the reference that Tropp uses for the earliest analysis of SRHT, although I do agree it does not seem correct.